WAW.API/Chat/Services/MessageService.cs

```diff
@@ -0,0 +1,45 @@
+using WAW.API.Chat.Domain.Models;
+using WAW.API.Chat.Domain.Repositories;
+using WAW.API.Chat.Domain.Services;
+using WAW.API.Chat.Domain.Services.Communication;
+using WAW.API.Shared.Domain.Repositories;
+
+namespace WAW.API.Chat.Services;
+
+public class MessageService : IMessageService {
+  private readonly IMessageRepository repository;
+  private readonly IUnitOfWork unitOfWork;
+
+  public MessageService(IMessageRepository repository, IUnitOfWork unitOfWork) {
+    this.repository = repository;
+    this.unitOfWork = unitOfWork;
+  }
+
+  public Task<IEnumerable<Message>> ListAll() {
+    return repository.ListAll();
+  }
+
+  public async Task<MessageResponse> Create(Message message) {
+    try {
+      await repository.Add(message);
+      await unitOfWork.Complete();
+      return new MessageResponse(message);
+    } catch (Exception e) {
+      return new MessageResponse($"An error occurred while saving the message: {e.Message}");
+    }
+  }
```

Comment:
I'll just leave this here for future reference: during message creation, the `SenderId` should be auto populated from the currently logged in user. Also, the controller should check if the current user is a participant of the requested `ChatRoom`. If they are not a participant, the message creation should be rejected.